pubsub/google/cloud/pubsub_v1/publisher/_batch/thread.py

```diff
@@ -148,6 +147,23 @@ def status(self):
         """
         return self._status
 
+    def cancel(self):
```

Comment:
It would be nice to actually pass in a message regarding why cancelation occurred. When it is due to a failure to publish an earlier message for an ordering key, we should surface that to the user.

Reply:
Done.